backend/app/compat.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime
+from typing import Optional
+
+import requests
+from fastapi import APIRouter, FastAPI
+
+from . import apps, db, search
+
+router = APIRouter(prefix="/compat")
+
+
+def register_to_app(app: FastAPI):
+    app.include_router(router)
+
+
+def get_repo_creation_date(appid: str) -> Optional[str]:
+    if created_at := db.redis_conn.get(f"{appid}:added_at"):
+        return created_at
+    else:
+        try:
+            github_repo = requests.get(f"https://api.github.com/repos/flathub/{appid}")
```

Comment:
This might lead to us exceeding the api limit, but hopefully not for long until everything is filled.